server/foodAPI/IFoodAPI.ts

```diff
@@ -0,0 +1,32 @@
+import IBaseFood from "../serverAPI/model/food/IBaseFood";
+import IFood from "../serverAPI/model/food/IFood";
+
+export default interface IFoodAPI {
+    /**
+     * Retrieves collection of food items that satisfy searching parameters.
+     * 
+     * @param parameters query parameters used for searching.
+     * - query - required parameter that defines the name of the Food Item (partial names are accepted).
+     * - number - optional parameter that defines max numbe of the results to be returned.
+     * - intolerence - optional parameter that defines the type of intolerences to be taken in consideration during searching.
+     * Complete list of intolerences is available at https://spoonacular.com/food-api/docs#Intolerances 
+     * 
+     * @throws NoParameterFound exception when required parameters weren't found.
+     * @returns Promise filled with an array of IFood objects.
+     */
+    SearchFood(parameters: Map<string, any>): Promise<IBaseFood[]>;
+
+    /**
+     * Retrieves food item that is specified by searching parameters.
+     * 
+     * @param parameters query parameters used for searching.
+     * - id - required parameter that defines unique identifier of the Food Item.
+     * - amount - optional parameter that defines max number of the food items. (default = 1)
```

Comment:
Yeah, GetFood returns specific item info. SearchFood is for searching when you don't know the id of the item.

Reply:
honestly the comment is misleading. Amount specify the amount of that specific food. The nutrition facts are based on food amount.

Reply:
We discussed offline and realized we need `number` and `amount` for all of these types of requests per the spoonacular [documentation](https://spoonacular.com/food-api/docs#Get-Ingredient-Information)

Reply:
Implemented in the update-authentication branch... instead of number it has full pagination with parameters "page" and "resultsPerPage" (for all 3rd party endpoints that return array of results). amount rename to "quantity" and support for "unit" has been added to the appropriate endpoint.